Honeycomb Bravo.lua

```diff
@@ -116,396 +133,617 @@ function array_has_true(array)
+	-- canopy = dataref_table('sim/cockpit/switches/canopy_open')
+	-- doors = dataref_table('sim/cockpit/switches/door_open')
+	cabin_door = dataref_table('AirbusFBW/PaxDoorArray')
+	doors_array[0] = dataref_table('AirbusFBW/PaxDoorArray')
+	doors_array[1] = dataref_table('AirbusFBW/CargoDoorArray')
+	doors_array[2] = dataref_table('AirbusFBW/BulkDoor')
+elseif PROFILE == 'DH8D' then
+	volt_low = dataref_table('sim/cockpit2/electrical/battery_voltage_actual_volts')
+	volt_low_min = 20
+	cabin_door = dataref_table('FJS/Q4XP/Manips/CabinMainDoor_Anim')
+	doors_array[0] = dataref_table('FJS/Q4XP/Manips/CabinDoors_Anim')
+	doors_array[1] = dataref_table('FJS/Q4XP/Manips/CargoDoor_Anim')
+end
+
+local DOOR_LAST_FLASHING = -1
+local DOOR_LAST_STATE = true
+
+function handle_led_changes()
+	if bus_voltage[0] > 0 or array_has_positives(bus_voltage) then
+		master_state = true
+
+		-- HDG & NAV
+		if bitwise.band(autopilot_state[0], 2) > 0 then
+			-- Heading Select Engage
+			set_led(LED.FCU_HDG, true)
+			set_led(LED.FCU_NAV, false)
+		elseif bitwise.band(autopilot_state[0], 512) > 0 or bitwise.band(autopilot_state[0], 524288) > 0 then
+			-- Nav Engaged or GPSS Engaged
+			set_led(LED.FCU_HDG, false)
+			set_led(LED.FCU_NAV, true)
+		elseif ONLY_USE_AUTOPILOT_STATE then
+			-- Aircraft known to use autopilot_state
+			set_led(LED.FCU_HDG, false)
+			set_led(LED.FCU_NAV, false)
+		else
+			if PROFILE == "Toliss/32x" then
+				if int_to_bool(ap1[0]) or int_to_bool(ap2[0]) then
+					if hdg[0] == 101 then
+						set_led(LED.FCU_HDG, true)
+						set_led(LED.FCU_NAV, false)
+					else
+						set_led(LED.FCU_HDG, false)
+						set_led(LED.FCU_NAV, true)
+					end
+				else
+					set_led(LED.FCU_HDG, false)
+					set_led(LED.FCU_NAV, false)
+				end
 			else
 				-- HDG
 				set_led(LED.FCU_HDG, get_ap_state(hdg))
 
 				-- NAV
 				set_led(LED.FCU_NAV, get_ap_state(nav))
 			end
+		end
+
+		-- APR
+		set_led(LED.FCU_APR, get_ap_state(apr))
 
-			-- APR
-			set_led(LED.FCU_APR, get_ap_state(apr))
+		-- REV
+		set_led(LED.FCU_REV, get_ap_state(rev))
 
-			-- REV
-			set_led(LED.FCU_REV, get_ap_state(rev))
+		-- ALT
+		local alt_bool
 
-			-- ALT
-			local alt_bool
+		if alt[0] > 1 then
+			alt_bool = true
+		else
+			alt_bool = false
+		end
 
-			if alt[0] > 1 then
-				alt_bool = true
+		if PROFILE == "Toliss/32x" then
+			if alt[0] == 0 or (not int_to_bool(ap1[0]) and not int_to_bool(ap2[0])) then
+				set_led(LED.FCU_ALT, false)
 			else
-				alt_bool = false
+				set_led(LED.FCU_ALT, true)
 			end
-
+		else
 			set_led(LED.FCU_ALT, alt_bool)
+		end
 
-			-- VS
-			set_led(LED.FCU_VS, get_ap_state(vs))
-
-			-- IAS
-			if bitwise.band(autopilot_state[0], 8) > 0 then
-				-- Speed-by-pitch Engage AKA Flight Level Change
-				-- See "Aliasing of Flight-Level Change With Speed Change" on https://developer.x-plane.com/article/accessing-the-x-plane-autopilot-from-datarefs/
-				set_led(LED.FCU_IAS, true)
+		-- VS
+		if PROFILE == "Toliss/32x" then
```

Comment:
The above suggests that we should probably consider the same thing, e.g. a `vs_bool` approach here. Or take a look at the overall approach for this different logic.